File: examples/lua53/client/protocols/AckRoleRandNameOk.lua
local M = {}
local moduleName = ...
_G[moduleName] = M
setmetatable(M, {__index = _G})
local _ENV = M


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckRoleRandNameOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uname = 0
end


function decode(self, pack)
	self.uname = pack:readString()
	return self
end


function getUname(self)
	return self.uname
end

File: examples/javascript/client/proto/AckRoleRandNameOk.js
module.exports = AckRoleRandNameOk;

var Packet = require('../net/Packet');


function AckRoleRandNameOk() {
	this._uname = undefined;


	this.Decode = function(packet) {
		this._uname = packet.ReadString();
	}


	this.SetUname = function(uname) {
		this._uname = uname;
	}
	this.GetUname= function() {
		return this._uname;
	}
}
